### PR TITLE
Remove summarizer, ground truth deployment, and Ray serve

### PR DESCRIPTION
**Pull request by veekaybee**
## What's changing

We're working towards migrating from Ray Serve to offline inference via vLLM for our ground truth generation. This involves three moving pieces: 

1. Getting rid of summarizer deployment and all related ground truth service assets since we won't be serving the model as a stand-alone web application but running in-process batch async evaluation.  **<-- this PR**
2. Moving `experiments` logically to jobs since we now have granular inference and evaluation jobs instead of full-length evaluation experiments. 
3. Creating a new module, `inference`, that allows for only the generation of ground truth in the same way as an evaluation job. 

**I won't be merging this PR until the other ground truth functionality is ready and working, including tests, but wanted to get it approved in small chunks** 

## How to test it

You should no longer see a /groundtruth endpoint and not be able to hit it via cURL
Screenshot 2024-10-15 at 11 33 38 AM

## I already...

- [X] added some tests for any new functionality - removed tests for deployment components
- [X] updated the documentation.

cc @javiermtorres @peteski22 FYI

### CODE REVIEW

File: lumigator/infra/mzai/helm/lumigator/values.yaml [hunk 1]
```diff
@@ -22,7 +22,6 @@ rayWorkerGPUs: ""
 
 serviceAccountName: ""
 
-summarizerWorkDir: ""
```
Comment (veekaybee) [inline lumigator/infra/mzai/helm/lumigator/values.yaml:25]: @chainlink anything else that needs to be changed on the infra side?

**Comment by aittalam** [comment]
Thanks Vicki! I ran `ack` on the codebase and found the following extra occurrences of summarizer:

```
pants.toml
45:root_patterns = ["python/mzai/backend", "lumigator/python/mzai/sdk", "lumigator/python/mzai/summarizer", "infra"]

README.md
32:  + Online Evaluations with [Ray Serve Deployments](lumigator/python/mzai/summarizer/README.md)

.env
1:PYTHONPATH=./lumigator/infra:./lumigator/python/mzai/backend:./lumigator/python/mzai/sdk:./lumigator/python/mzai/summarizer:$PYTHONPATH

lumigator/infra/mzai/helm/lumigator/README.md
59:| summarizerWorkDir | `""` | S3-compatible storage system folder where the summarizer is located |

lumigator/infra/mzai/helm/lumigator/templates/deployment.yaml
65:              value: {{ required "Missing Summarizer work directory name" .Values.summarizerWorkDir | quote }}

notebooks/lumigator_demo.py
316:def get_summarizer_deployment_id():
317:    """Gets the summarizer deployment id from ray serve."""
318:    path = "applications/summarizer/deployed_app_config/args/description"

notebooks/walkthrough.ipynb
486:    "deployment_id = ld.get_summarizer_deployment_id()\n",
```

Perhaps a few of these are lower priority (e.g. we can wait to update the notebook when we have a mean to calculate GT), but I hope some of these pointers can be actually useful to you

File: README.md [hunk 2]
```diff
@@ -73,10 +72,9 @@ See [example notebook](/notebooks/walkthrough.ipynb) for a platform API walkthro
 Lumigator is a Python-based FastAPI web app with REST API endpoints that allow for access to services for serving and evaluating large language models available as safetensor artifacts hosted on both HuggingFace and local stores, with our first primary focus being Huggingface access, and tracking the lifecycle of a model in the backend database (Postgres).
 It consists of:
 
-+ a FastAPI-based huggingface's `evaluate` library for those metrics, but we are considering using lm-harness that manages platform activity backed by Postgres
-+ online evaluation of models using **Ray Serve** deployments
-+ a **Ray cluster** to run offline evaluation jobs using [lm-buddy](https://github.com/mozilla-ai/lm-buddy), our in-house eval framework
-    + LM buddy runs inference accessing different kind of models, accessible locally or via APIs, and evaluation with huggingface's `evaluate` library or lm-evaluation-harness
++ a FastAPI-based web app that includes  huggingface's `evaluate` library for those metrics
```
Comment (veekaybee) [inline README.md:75]: Updated these docs also to remove lm-buddy

File: lumigator/python/mzai/sdk/tests/data/deployments.json
Comment (veekaybee) [inline lumigator/python/mzai/sdk/tests/data/deployments.json:1]: @peteski22 FYI

**Comment by veekaybee** [comment]
Thanks @aittalam! I also ran git-grep after this and found even more instances, so probably my lesson is don't use PyCharm find functionality moving forward.  They should all be gone now, except for the notebook and `.env`, which is a local file we [ignore via .gitignore.](https://github.com/mozilla-ai/lumigator/blob/0d7ea49b01a9ba8b41697d3fc3f42b50a41cc003/.gitignore#L134) 

> Thanks Vicki! I ran `ack` on the codebase and found the following extra occurrences of summarizer:
> 
> ```
> pants.toml
> 45:root_patterns = ["python/mzai/backend", "lumigator/python/mzai/sdk", "lumigator/python/mzai/summarizer", "infra"]
> 
> README.md
> 32:  + Online Evaluations with [Ray Serve Deployments](lumigator/python/mzai/summarizer/README.md)
> 
> .env
> 1:PYTHONPATH=./lumigator/infra:./lumigator/python/mzai/backend:./lumigator/python/mzai/sdk:./lumigator/python/mzai/summarizer:$PYTHONPATH
> 
> lumigator/infra/mzai/helm/lumigator/README.md
> 59:| summarizerWorkDir | `""` | S3-compatible storage system folder where the summarizer is located |
> 
> lumigator/infra/mzai/helm/lumigator/templates/deployment.yaml
> 65:              value: {{ required "Missing Summarizer work directory name" .Values.summarizerWorkDir | quote }}
> 
> notebooks/lumigator_demo.py
> 316:def get_summarizer_deployment_id():
> 317:    """Gets the summarizer deployment id from ray serve."""
> 318:    path = "applications/summarizer/deployed_app_config/args/description"
> 
> notebooks/walkthrough.ipynb
> 486:    "deployment_id = ld.get_summarizer_deployment_id()\n",
> ```
> 
> Perhaps a few of these are lower priority (e.g. we can wait to update the notebook when we have a mean to calculate GT), but I hope some of these pointers can be actually useful to you

Thanks @aittalam! I ran `git grep` on this as well afterwards and found even more references, so lesson learned here not to use PyCharm for find and replace alone. I removed everything except the mentions in the notebook for now and the `.env` file since that should be local and [in .gitignore.](https://github.com/mozilla-ai/lumigator/blob/0d7ea49b01a9ba8b41697d3fc3f42b50a41cc003/.gitignore#L134)

File: lumigator/infra/mzai/helm/lumigator/values.yaml (hunk 1 above)
Comment (chainlink) [reply lumigator/infra/mzai/helm/lumigator/values.yaml:25]: Looks good!

**Review by chainlink — APPROVED** [review]
LGTM!

**Review by aittalam — APPROVED** [review]
🚀

**Pull request merged**